src/test/ui/nll/issue-58053.rs

```diff
@@ -0,0 +1,9 @@
+#![allow(warnings)]
+#![feature(nll)]
+
+fn main() {
+    let f = |x: &i32| -> &i32 { x };
+    //~^ ERROR lifetime may not live long enough
+    let i = &3;
+    let j = f(i);
```

Comment:
Add a test for the no-return-type case as well.

Reply:
Added.